### PR TITLE
tldr. detailed added in metadata of annotations tag in YAML

### PR DESCRIPTION
**Pull request by aloksharma20**
**Purpose of PR?**:

Showing tldr and detailed for hardening policies 

Fixes #

Added tldr and detailed while creating hardening policies and added  inside metadata of policy with annotation tag

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: src/plugin/kubearmor.go [hunk 1]
```diff
@@ -49,14 +50,20 @@ func ConvertKnoxSystemPolicyToKubeArmorPolicy(knoxPolicies []types.KnoxSystemPol
 		kubePolicy := types.KubeArmorPolicy{
 			APIVersion: "security.kubearmor.com/v1",
 			Kind:       "KubeArmorPolicy",
-			Metadata:   map[string]string{},
+			Metadata: metav1.ObjectMeta{
+				Name:        "",
+				Namespace:   "",
+				Annotations: map[string]string{},
```
Comment (avithe-great) [inline src/plugin/kubearmor.go]: As  `Setting Fields After Declaration` approach is better in this case. can be done in this way.

```
kubePolicy := types.KubeArmorPolicy{
	APIVersion: "security.kubearmor.com/v1",
	Kind:       "KubeArmorPolicy",
}

// Set fields after declaration
kubePolicy.Metadata.Name = xxxxxxxxx
kubePolicy.Metadata.Annotations = map[string]string{
	"tldr":        "zzzzzz",
	"description": "yyyyyy",
}
```
By not providing an empty ObjectMeta{} during declaration, we avoid unnecessary initialization of the Metadata field, which makes the code slightly more efficient. When we set the fields later, Go will automatically initialize Metadata as an empty ObjectMeta{} when needed.

File: src/systempolicy/systemPolicy.go [hunk 2]
```diff
@@ -271,14 +271,14 @@ func populateKnoxSysPolicyFromWPFSDb(namespace, clustername, labels, fromsource
 func WriteSystemPoliciesToFile_Ext(namespace, clustername, labels, fromsource string, includeNetwork bool) {
 	kubearmorK8SPolicies := extractK8SSystemPolicies(namespace, clustername, labels, fromsource, includeNetwork)
 	for _, pol := range kubearmorK8SPolicies {
-		fname := "kubearmor_policies_" + pol.Metadata["clusterName"] + "_" + pol.Metadata["namespace"] + "_" + pol.Metadata["containername"] + "_" + pol.Metadata["name"]
+		fname := "kubearmor_policies_" + pol.Metadata.Annotations["clusterName"] + "_" + pol.Metadata.Annotations["namespace"] + "_" + pol.Metadata.Annotations["containername"] + "_" + pol.Metadata.Annotations["name"]
```
Comment (avithe-great) [inline src/systempolicy/systemPolicy.go]: here i feel that it (namespaces,name etc) should be part of Metadata spec not annotation spec but i would suggest to check with @wazir-ahmed

**Review by avithe-great — CHANGES_REQUESTED** [review]

File: src/recommendpolicy/helperFunctions.go [hunk 3]
```diff
@@ -326,12 +326,16 @@ func createPolicy(ms types.MatchSpec, name, namespace string, labels LabelMap) (
 
 	if policy.Kind != types.KindKubeArmorHostPolicy {
 		policy.Metadata = map[string]string{
-			"name":      fmt.Sprintf("%v-%v-%v", types.HardeningPolicy, name, ms.Name),
-			"namespace": namespace,
+			"name":            fmt.Sprintf("%v-%v-%v", types.HardeningPolicy, name, ms.Name),
+			"namespace":       namespace,
+			"annotation-tldr": ms.Description.Tldr,
+			"annotation-desc": ms.Description.Detailed,
 		}
```
Comment (avithe-great) [inline src/recommendpolicy/helperFunctions.go]: Can be minimised-
```
// Set metadata fields based on policy.Kind
	policy.Metadata = map[string]string{
		"annotation-tldr": ms.Description.Tldr,
		"annotation-desc": ms.Description.Detailed,
	}
	if policy.Kind != types.KindKubeArmorHostPolicy {
		policy.Metadata["name"] = fmt.Sprintf("%v-%v-%v", types.HardeningPolicy, name, ms.Name)
		policy.Metadata["namespace"] = namespace
	} else {
		policy.Metadata["name"] = fmt.Sprintf("%v-host-%v-%v", types.HardeningPolicy, name, ms.Name)
	}
```

**Review by avithe-great — CHANGES_REQUESTED** [review]

File: src/plugin/kubearmor.go (hunk 1 above)
Comment (aloksharma20) [reply src/plugin/kubearmor.go]: addressed

File: src/systempolicy/systemPolicy.go (hunk 2 above)
Comment (aloksharma20) [reply src/systempolicy/systemPolicy.go]: addressed

File: src/recommendpolicy/helperFunctions.go (hunk 3 above)
Comment (aloksharma20) [reply src/recommendpolicy/helperFunctions.go]: addressed

**Review by avithe-great — APPROVED** [review]

File: src/systempolicy/systemPolicy.go [hunk 4]
```diff
@@ -271,14 +271,14 @@ func populateKnoxSysPolicyFromWPFSDb(namespace, clustername, labels, fromsource
 func WriteSystemPoliciesToFile_Ext(namespace, clustername, labels, fromsource string, includeNetwork bool) {
 	kubearmorK8SPolicies := extractK8SSystemPolicies(namespace, clustername, labels, fromsource, includeNetwork)
 	for _, pol := range kubearmorK8SPolicies {
-		fname := "kubearmor_policies_" + pol.Metadata["clusterName"] + "_" + pol.Metadata["namespace"] + "_" + pol.Metadata["containername"] + "_" + pol.Metadata["name"]
+		fname := "kubearmor_policies_" + "_" + pol.Metadata.Namespace + "_" + "_" + pol.Metadata.Name
```
Comment (wazir-ahmed) [inline src/systempolicy/systemPolicy.go]: There are two `_` between `Namespace` & `Name`

File: src/plugin/kubearmor.go [hunk 5]
```diff
@@ -49,14 +49,17 @@ func ConvertKnoxSystemPolicyToKubeArmorPolicy(knoxPolicies []types.KnoxSystemPol
...
 		if policy.Kind != types.KindKubeArmorHostPolicy {
-			kubePolicy.Metadata["namespace"] = policy.Metadata["namespace"]
+			kubePolicy.Metadata.Namespace = policy.Metadata["namespace"]
 		}
 
-		kubePolicy.Metadata["name"] = policy.Metadata["name"]
+		kubePolicy.Metadata.Name = policy.Metadata["name"]
+		kubePolicy.Metadata.Annotations = map[string]string{
+			"tldr":        policy.Metadata["annotation-tldr"],
+			"description": policy.Metadata["annotation-description"],
```
Comment (wazir-ahmed) [inline src/plugin/kubearmor.go]: Use the following keys in annotation.
```
app.accuknox.com/tldr: <tldr>
app.accuknox.com/description: <description>
```

**Review by wazir-ahmed — CHANGES_REQUESTED** [review]

File: src/plugin/kubearmor.go (hunk 5 above)
Comment (aloksharma20) [reply src/plugin/kubearmor.go]: addressed

File: src/systempolicy/systemPolicy.go [hunk 6]
```diff
@@ -271,14 +271,14 @@ func populateKnoxSysPolicyFromWPFSDb(namespace, clustername, labels, fromsource
 func WriteSystemPoliciesToFile_Ext(namespace, clustername, labels, fromsource string, includeNetwork bool) {
 	kubearmorK8SPolicies := extractK8SSystemPolicies(namespace, clustername, labels, fromsource, includeNetwork)
 	for _, pol := range kubearmorK8SPolicies {
-		fname := "kubearmor_policies_" + pol.Metadata["clusterName"] + "_" + pol.Metadata["namespace"] + "_" + pol.Metadata["containername"] + "_" + pol.Metadata["name"]
+		fname := "kubearmor_policies_" + "_" + pol.Metadata.Namespace + "_" + pol.Metadata.Name
```
Comment (vishnusomank) [inline src/systempolicy/systemPolicy.go]: we have double `_` in this also

File: src/systempolicy/systemPolicy.go (hunk 4 above)
Comment (aloksharma20) [reply src/systempolicy/systemPolicy.go]: addressed

File: src/plugin/kubearmor.go [hunk 7]
```diff
@@ -49,14 +49,17 @@ func ConvertKnoxSystemPolicyToKubeArmorPolicy(knoxPolicies []types.KnoxSystemPol
...
-			kubePolicy.Metadata["namespace"] = policy.Metadata["namespace"]
+			kubePolicy.Metadata.Namespace = policy.Metadata["namespace"]
 		}
 
-		kubePolicy.Metadata["name"] = policy.Metadata["name"]
+		kubePolicy.Metadata.Name = policy.Metadata["name"]
+		kubePolicy.Metadata.Annotations = map[string]string{
+			"app.accuknox.com/tldr":        policy.Metadata["annotation-tldr"],
+			"app.accuknox.com/description": policy.Metadata["annotation-description"],
+		}
```
Comment (vishnusomank) [inline src/plugin/kubearmor.go]: can we check to see if either of this fields  (' annotation-tldr` or `annotation-description`) are empty and then add an annotation 

```suggestion
		if policy.Metadata["annotation-tldr"] != "" || policy.Metadata["annotation-description"] != "" {
			kubePolicy.Metadata.Annotations = map[string]string{
				"app.accuknox.com/tldr":        policy.Metadata["annotation-tldr"],
				"app.accuknox.com/description": policy.Metadata["annotation-description"],
			}
		}
```

**Review by vishnusomank — CHANGES_REQUESTED** [review]
For system policies, the annotations will always be empty. So shall we have a check

File: src/systempolicy/systemPolicy.go (hunk 6 above)
Comment (aloksharma20) [reply src/systempolicy/systemPolicy.go]: addressed

File: src/plugin/kubearmor.go (hunk 7 above)
Comment (aloksharma20) [reply src/plugin/kubearmor.go]: addressed

**Review by vishnusomank — APPROVED** [review]
LGTM!

**Review by wazir-ahmed — APPROVED** [review]

**Pull request merged**